components/tools/OmeroPy/src/omero/gateway/__init__.py

```diff
@@ -2324,6 +2324,19 @@ def isAdmin(self):
 
         return self.getEventContext().isAdmin
 
+    def isFullAdmin(self):
+        """
+        Checks if a user has full administration privileges.
+
+        :return:    Boolean
+        """
+
+        if self.getEventContext().isAdmin:
+            allPrivs = list(self.getEnumerationEntries('AdminPrivilege'))
```

Comment:
Would `allPrivs` be worth caching, or noting upon connection?

Reply:
I guess it depends how expensive it is to ```getEnumerationEntries()```. I don't imagine you'd be calling this multiple times in normal usage?

Reply:
Fair enough, not much more expensive than the server roundtrip overhead. Not a blocker.